include/gsl/span

```diff
@@ -517,10 +516,13 @@ private:
     class storage_type : public ExtentType
     {
     public:
+
+        // basechecked needed to remove unnecessary null check in subspans
```

Comment:
Minor thing: could we change the name of "basechecked" to something more descriptive. Maybe "null_checked" or just "checked". It's the "base" part I find confusing.

Reply:
done!